tests/integration/conftest.py

```diff
@@ -41,8 +65,12 @@
 def shell(tmp_path):
     """Live shell."""
 
-    def factory(command, silent=False):
-        if not silent:
+    def factory(command, silent=False, hide_command=False):
```

Comment:
What is the difference between silent and hide_command? Seems to me that they will do the same thing in the code

Reply:
`silent` is for no output from the shell fixture at all, apart from printing, that no output will be provided.
That might be used for some calls where we don't want the logs too polluted, for example when we install/uninstall packages and the output gets quite lengthy.
`hide_command` serves the purpose of trying not to leak any sensitive data. For example, if we call `subscription-manager register` directly, the shell fixture prints out the used command with the credentials in clear text first and then output of the running command.
In that case, we might care about the output for debugging purposes, but we don't want to show the command used (that you can find easily in the code if needed).

Reply:
What we would like to do in the future is to use logger and log a respective level based on the shell call returncode.
So if everything runs fine, don't print out anything, if there is a problem, fallback to debug/error/critical and print out the output.